sharded_queue/api.lua

```diff
@@ -402,8 +422,142 @@ function sharded_tube.drop(self)
 end
 
 local sharded_queue = {
-    tube = {}
+    tube = {},
+    cfg = stash_get(stash_names.cfg),
+    metrics_stats = stash_get(stash_names.metrics_stats),
 }
+if sharded_queue.cfg.metrics == nil then
+    sharded_queue.cfg.metrics = true
+end
+
+local function is_metrics_v_0_11_installed()
+    if not is_metrics_package or metrics.unregister_callback == nil then
+        return false
+    end
+    local counter = require('metrics.collectors.counter')
+    return counter.remove and true or false
+end
+
+local function metrics_create_collectors()
+    local create_counter = function(name, description)
+        return {
+            collector = metrics.counter(name, description),
+            values = {},
+        }
+    end
+    local create_gauge = function(name, description)
+        return {
+            collector = metrics.gauge(name, description)
+        }
+    end
+    local call_names = {
+        "done",
+        "take",
+        "kick",
+        "bury",
+        "put",
+        "delete",
+        "touch",
+        "ack",
+        "release",
+    }
+
+    local calls = {}
+    for _, name in ipairs(call_names) do
+        calls[name] = create_counter(
+            "sharded_queue_calls_" .. name,
+            "sharded_queue's number of '" .. name .. "' calls"
+        )
+    end
+
+    local tasks = {}
+    for name, _ in pairs(state) do
+        local lower = name:lower()
+        tasks[lower] = create_gauge(
+            "sharded_queue_tasks_" .. lower,
+            "sharded_queue's number of '" .. lower .. "' tasks"
+        )
+    end
+    tasks["total"] = create_gauge(
+        "sharded_queue_tasks_total",
```

Comment:
On the other hand, having `_total` in a gauge is treated as an error. It wouldn't pass Prometheus validation with `promtool`. So it is either a counter or shouldn't be named `*_total`.

Reply:
`_all` ?

Reply:
total/all tasks = sum of all tasks by types (count of  ready tasks + count of taken tasks etc.)

Reply:
No, it is not. A task could be deleted from a tube (on 'ack'/'delete' calls).

Reply:
> `active`

Maybe this one is better suited for our case

Reply:
I think it's fine.